### ISSUE TITLE
ELK 6 

### ISSUE THREAD

**Issue by buenoser** [tuckner/f5-elk#1]
Hi,

I test to do it with ELK 6 and F5 BigIP 13.0, but nothing works anymore.

Have you test it with this conponents ?

**Comment by tuckner** [comment]
I haven't tested with ELK 6.  I believe some of the type mappings in the template may have been depreciated and Kibana changes often as far as dashboard widgets are concerned, but as far as I know, the F5 log format should be the same (key/value) and Logstash should still be able to process them into Elastic

**Comment by buenoser** [comment]
Hi,
For the ASM logs, it's worked. In fact the mapping dosen't work with the template.

And the automatic mapping do like this: 

<131>Jul 27 15:55:09 F5DEVICE.domain.lu ASM:unit_hostname
<131>Jul 27 15:55:09 F5DEVICE.domain.lu ASM:unit_hostname.keyword

So 1 new entry = 2 new fields, i supposed it's because the template doesen't load.


The Config Logstash is like this: 


```
 input {
  udp {
    port => 1517
    type => 'hsl'
  }
  tcp {
    port => 1517
    type => 'syslog'
  }
}


filter {
  kv {
    field_split => ","
    trim_value => "\""
  }
  if ([ClientIP]) {
      geoip {
        source => "ClientIP"
    }
  }

}

output {
  stdout {}

  elasticsearch {
    hosts => ["ELK-TEST07-VMR:9200"]
    index => "logstash-%{+YYYY.MM.dd}"
    template => "/etc/logstash/templates/f5-template.json"
    template_name => "f5-template"
    template_overwrite => true
  }
}

```


Regards

**Comment by buenoser** [comment]
In the Big IP 13.X, The application Security propose this: 
![image](https://user-images.githubusercontent.com/41565842/43327011-8fe68730-91ba-11e8-8134-9d9a29aea8d6.png)


In Logging Format:
![image](https://user-images.githubusercontent.com/41565842/43327023-9aab9480-91ba-11e8-8971-95462b4dbe60.png)

If you want, I can help to update this great intégration with ELK 6.X and Big IP 13.X ?

**Comment by tuckner** [comment]
Yeah!  Submit a pull or send me what should be updated and I'll look it over!

**Comment by buenoser** [comment]
Great, I will do this for ASM! 

For LTM, I have an issue, I follow your post, and nothing is ingested, 
I configured a simple VS and aply an Analytic profil with an Log Destination / Publisher, but nothing enter in ELK. What can I do to find the issue ?

Regards

**Comment by tuckner** [comment]
Can you verify F5 is sending the logs succesfully?  I seem to remember not being able to send syslog out of the management interface of the F5.

**Comment by buenoser** [comment]
Hi,
Thank you for the response,
I folloed this instruction https://support.f5.com/csp/article/K13080 and it's working for syslog ! But the LTM from AVR nothing appear, how can i check if its really send?

Regards

**Comment by tuckner** [comment]
Make sure to apply the AVR profile to your virtual server, have the correct pool of AVR logging servers, and make sure you're not sending logs out the management port.  Other than that, maybe using tcpdump will help validate if the logs are actually being sent or not.

**Comment by buenoser** [comment]
Hi, I understand what you mean, but I don't know how to check if the AVR use the management port to send this logs... 

Can you help me to check this ?

**Comment by tuckner** [comment]
It would depend on routing.  If the fastest path to the ELK server is through the management port, then AVR will probably fail to send the logs.
